docs/DeveloperGuide.md

```diff
@@ -154,10 +154,87 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+![AddPersonSequenceDiagram](images/AddPersonSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** `cmd` in the diagram represents the add 
+person command text entered by user. The specific UniquePersonList operations are not shown in the diagram
+for simplicity.
+</div>
+
+The following activity diagram shows what happens when a user executes a new add command:
+
+![AddPersonActivityDiagram](images/AddPersonActivityDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Only the 
+activities related to gender field is considered and shown in this activity diagram.
+</div>
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:**Parser exceptions is thrown and caught if 
+gender field in not provided in the command, or the gender is not of valid format; Duplicated person exception is 
+thrown if the person to add already exists in the contact list. Error message is displayed on the GUI subsequently.
+</div>
+
+#### Design considerations:
+
+**Aspect: Whether gender field should be optional for a person:**
+
+* **Alternative 1 (current choice):** Compulsory gender field:
+    * Pros: It is a more logical implementation because gender is a common attribute for all persons,
+  similar to name, address, etc, which are also compulsory field for persons in contact list.
+    * Cons: It is less flexible, since only female and male genders are accepted.
+
+* **Alternative 2:** Optional gender field:
+    * Pros: It is a more flexible implementation, since user has the choice to set gender to male or female, 
+  as well as hide gender.
+    * Cons: It is less logical since gender is usually a required binary field in most applications.
+
+### Edit Gender
+
+The Edit Gender feature allows users to edit a gender field (Male / Female) of a person in the contact list. 
+It is performed as a part of `editCommand#execute()`.
+
+These operations are exposed in the `Model` interface as the method `Model#setPerson()`, which calls
+`AddressBook#setPerson()` which calls `UniquePersonList#setPerson` to replace an existing person with a new person 
+object with edited fields in the person list stored in AddressBook.
+
+The following sequence diagram shows the methods calls related to edit person operation:
+
+![EditPersonSequenceDiagram](images/EditPersonSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** `cmd` in the diagram represents the edit 
+person command text entered by user; the setPerson(P1, P2) method replaces person P1 with person P2 in the person list 
+in the model. The specific UniquePersonList operations are not shown in the diagram for simplicity.
+</div>
+
+The following activity diagram shows what happens when a user executes a new edit command:
+
+![EditPersonActivityDiagram](images/EditPersonActivityDiagram.png)
```

Comment:
Same issue as the add person activity diagram

Reply:
Same issue as the add person activity diagram, the branch node should be changed